pkg/autodiscover/autodiscover_pods.go

```diff
@@ -19,36 +19,62 @@ package autodiscover
 import (
 	"context"
 
+	"github.com/hashicorp/go-multierror"
 	"github.com/test-network-function/cnf-certification-test/internal/log"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
-func findPodsByLabel(oc corev1client.CoreV1Interface, labels []labelObject, namespaces []string) (runningPods, allPods []corev1.Pod) {
+func findAllPods(labelSelector map[string]string, namespace string, oc corev1client.CoreV1Interface) (*corev1.PodList, error) {
+	var allPods *corev1.PodList
+	var err error
+	allPods = &corev1.PodList{}
+	for key, value := range labelSelector {
+		log.Debug("Searching Pods in namespace %s with label selector %q", namespace, ""+key+"="+value)
+		pods, errList := oc.Pods(namespace).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: key + "=" + value,
+		})
+		if errList != nil {
+			log.Error("Error when listing pods in ns=%s, err: %v", namespace, errList)
+			err = multierror.Append(err, errList)
+		}
+		allPods.Items = append(allPods.Items, pods.Items...)
+	}
+	return allPods, err
+}
+func findPodsByLabel(oc corev1client.CoreV1Interface, labels []labelObject, namespaces []string) (runningPods, allPods []corev1.Pod, err error) {
```

Comment:
I would rename this function findPodsByLabel to findPodsUnderTest (same for pods, deployment, statefulset and operator)

Reply:
Looking at this again, I would keep the initial name because we are using the findPodsByLabel function also to get the debug pods, which are not pods under test. I would just add a "s" to account for us looking for multiple labels, so findPodsByLabel -> findPodsByLabels